packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/data_query/data_query_type_check.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    components::ComponentVariantPropTypes,
+    props::{prop_profile_to_type, PropProfile},
+};
+
+use super::{DataQuery, PropSpecifier, PropValueType};
+
+impl DataQuery {
+    /// Guess the return type of a data query. Since this return type may depend on
+    /// retrieving a prop from a component, you must pass in a reference to the component.
+    ///
+    /// Note: this function assumes the data query is well formed (E.g., it only uses `PropSpecifier::LocalIdx` in conjunction
+    /// with `PropSource::Me`). If the data query is not well formed, this function may return an incorrect type.
+    #[allow(clippy::result_unit_err)]
+    pub fn guess_return_type<T: ComponentVariantPropTypes + 'static>(
```

Comment:
If I understand this correctly, it used only in tests. If so, then include text to mention that, and maybe start with an `_` like the other functions.